adapters/zio-http/src/main/scala/caliban/ZHttpAdapter.scala

```diff
@@ -50,11 +48,14 @@ object ZHttpAdapter {
                      case Right(output) => WebSocketFrame.Text(outputCodec.encode(output))
                      case Left(close)   => WebSocketFrame.Close(close.code, Some(close.reason))
                    }
-          socket = Http
-                     .collectZIO[WebSocketChannelEvent] {
-                       case ChannelEvent(ch, UserEventTriggered(HandshakeComplete)) =>
-                         out.runForeach(ch.writeAndFlush(_)).race(ch.awaitClose)
-                       case ChannelEvent(_, ChannelRead(WebSocketFrame.Text(text))) =>
+          socket = Handler.webSocket { ch =>
+                     ch.receiveAll {
+                       case ChannelEvent.UserEventTriggered(UserEvent.HandshakeComplete) =>
+                         out
+                           .interruptWhen(ch.awaitShutdown)
+                           .runForeach(msg => ch.send(ChannelEvent.Read(msg)))
+                           .forkDaemon
```

Comment:
I could only get this to work by forking this effect. Not sure if this is the right approach, any thoughts @frekw?

Reply:
I haven't had the time to dive into the new WebSocket APIs yet unfortunately :/

But I agree that this `forkDaemon` feels off. In general I'm not super fond of our setup where we create a separate queue for responses. I wonder if that contributes to the weirdness in this case, that we are adding more abstraction on top of zio-http's handling? 🤔